hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/security/RouterDelegationTokenSecretManager.java

```diff
@@ -91,283 +117,173 @@ private boolean shouldIgnoreException(Exception e) {
    * @param newKey DelegationKey
    */
   @Override
-  public void storeNewMasterKey(DelegationKey newKey) {
+  protected void storeNewMasterKey(DelegationKey newKey) throws IOException {
     try {
       federationFacade.storeNewMasterKey(newKey);
```

Comment:
Is the federationFacade being initialized with actual conf (that points to Sql state store let's say). Because FederationStateStoreFacade.getInstance(); will always get an instance that is initialized with default conf.

Reply:
yea thats fair. let me correct it